api/renewer_test.go

```diff
@@ -233,3 +237,60 @@ func TestLifetimeWatcher(t *testing.T) {
 		})
 	}
 }
+
+// randDuration calculates a random duration for use in property based testing.
+func randDuration(r *rand.Rand, max int64) time.Duration {
+	return time.Duration(max)
```

Comment:
@peteski22 any thoughts on how to better set the increment?

Reply:
Your recommendation to remove `randDuration` helped. All the inputs are related to each other, even though increment is a different type. (maybe something to fix in the LifeTimeWatcher).

It is way more straightforward to initialize them all as `r.Intn` and then convert them into Durations later.

---

Intuitively, I changed the `increment` to be less than the `remainingLeaseDuration`, but would want confirmation that that's correct.

Reply:
> Intuitively, I changed the `increment` to be less than the `remainingLeaseDuration`, but would want confirmation that that's correct.

I don't believe that the `increment` needs to be smaller than the duration, `increment` is the amount of time in seconds you want to extend your secret/token for, and so manifests as its `ttl` if all is well.

With the grace calculation, if we have an `increment` greater than `0`, then the lease duration is capped at that as the maximum duration (in `calcuateGrace` func).

Therefore I think this would be fine:
```
increment := r.Intn(leaseDuration + 1)
```